build/ci.go

```diff
@@ -345,6 +347,67 @@ func downloadSpecTestFixtures(csdb *build.ChecksumDB, cachedir string) string {
 	return filepath.Join(cachedir, base)
 }
 
+func hashFile(path string) (common.Hash, error) {
+	f, err := os.OpenFile(path, os.O_RDONLY, 0666)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	contents, err := io.ReadAll(f)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	return common.BytesToHash(crypto.Keccak256(contents)), nil
+}
+
+func generateCheck() {
+	tc := new(build.GoToolchain)
+	preHashes := make(map[string]common.Hash)
+	err := filepath.WalkDir(".", func(path string, d os.DirEntry, err error) error {
+		if strings.HasPrefix(path, "tests/testdata") {
+			return nil
+		}
+		if d.IsDir() {
+			return nil
+		}
+		preHash, err := hashFile(path)
+		if err != nil {
+			return err
+		}
+		preHashes[path] = preHash
+
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+	c := tc.Go("generate", "./...")
+	build.MustRun(c)
+	err = filepath.WalkDir(".", func(path string, d os.DirEntry, err error) error {
```

Comment:
It kind of feels like a `hashDirectory` function could be defined here? It's quite a bit of code you duplicated.